### PR TITLE
Add GoLink extension

### PR DESCRIPTION
**Pull request by micnncim**
## Description

Adds an extension for [GoLinks](https://golinks.io) with the following commands:

- Search golinks: Lets users search golinks and then open a selected golink.
- Create golink: Lets users create a new golink.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/21333876/141972649-5074a8d6-9af8-4392-a3b4-83545f185e17.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

File: extensions/golinks/package.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  "name": "golinks",
+  "title": "GoLinks",
+  "description": "Manage your golinks",
+  "icon": "command-icon.png",
+  "author": "micnncim",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "list",
+      "title": "Search golinks",
```
Comment (pernielsentikaer) [inline extensions/golinks/package.json:12]:
```suggestion
      "title": "Search Links",
```

File: extensions/golinks/package.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      "subtitle": "GoLinks",
+      "description": "Search golinks",
+      "mode": "view"
+    },
+    {
+      "name": "create",
+      "title": "Create golink",
```
Comment (pernielsentikaer) [inline extensions/golinks/package.json:19]:
```suggestion
      "title": "Create Link",
```

File: extensions/golinks/package.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      "subtitle": "GoLinks",
+      "description": "Create golink",
```
Comment (pernielsentikaer) [inline extensions/golinks/package.json:21]:
```suggestion
      "description": "Create a new GoLink",
```

File: extensions/golinks/package.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  "description": "Manage your golinks",
+  "icon": "command-icon.png",
+  "author": "micnncim",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "list",
+      "title": "Search golinks",
+      "subtitle": "GoLinks",
+      "description": "Search golinks",
```
Comment (pernielsentikaer) [inline extensions/golinks/package.json:14]:
```suggestion
      "description": "Search your GoLinks",
```

File: extensions/golinks/src/create.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      <Form.TextField id="name" title="Name" placeholder="Name" />
+      <Form.TextArea id="description" title="Description" placeholder="Description" />
+    </Form>
+  )
+}
+
+async function submit(values: Record<string, FormValue>) {
+  const body = { ...values }
+
+  showToast(ToastStyle.Animated, 'Creating golink...')
```
Comment (pernielsentikaer) [inline extensions/golinks/src/create.tsx:23]:
```suggestion
  showToast(ToastStyle.Animated, 'Creating GoLink...')
```

File: extensions/golinks/src/create.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import { ActionPanel, Form, FormValue, SubmitFormAction, popToRoot, showToast, ToastStyle } from '@raycast/api'
+import { createGolink } from './api'
+
+export default function CreateGolink() {
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <SubmitFormAction title="Create golink" onSubmit={submit} />
```
Comment (pernielsentikaer) [inline extensions/golinks/src/create.tsx:9]:
```suggestion
          <SubmitFormAction title="Create GoLink" onSubmit={submit} />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @micnncim  👋 

Interesting, I haven't heard about GoLink before, it looks useful in some scenarios 💪 

Beside the suggestions, I would like to see some kind of error handling when an invalid token is added.

When the user is visiting “Search Links” command, then the loader keeps spinning and no message about invalid token (as seen in “Create Link” when creating with invalid credentials)

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by micnncim** [comment]
@pernielsentikaer 

Thanks for your review!

> Beside the suggestions, I would like to see some kind of error handling when an invalid token is added.

Is it enough to validate just the length of a provided token before sending requests?

**Comment by pernielsentikaer** [comment]
> > Beside the suggestions, I would like to see some kind of error handling when an invalid token is added.
> 
> Is it enough to validate just the length of a provided token before sending requests?

I'm not familiar with the API, but when you use the “Search Links” command, then it's just showing the loading indicator and nothing happens (unlike the create command). Some adds an “error” item to the list like it's done in #165 

image

When it's only showing the spinner, then there is no way to find out what's wrong.

File: extensions/golinks/src/list.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import { ActionPanel, CopyToClipboardAction, Icon, List, OpenInBrowserAction } from '@raycast/api'
+import { useFetch } from './api'
+import { GoLink } from './types'
+
+export default function ListGoLink() {
+  const { data, isLoading, error } = useFetch('/golinks?limit=100')
+  if (error) {
+    return (
+      <List>
+        <List.Item icon={Icon.ExclamationMark} title="Error occurred" accessoryTitle={error.message} />
```
Comment (micnncim) [inline extensions/golinks/src/list.tsx:10]: Screenshot 2021-11-16 at 2 46 12

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @micnncim 👋 

Thanks for the update, it looks a lot better! I think I would show something like the screenshot if 401 is received since it means that provided token is wrong, for the rest of the errors I should show a toast — does it make sense?

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by micnncim** [comment]
@pernielsentikaer Fixed at [`4e62a56` (#346)](https://github.com/raycast/extensions/pull/346/commits/4e62a56fd58d83003b1924d47419a9a576e555b0). It shows an invalid token error in the `List` component and shows toasts for other errors.

**Comment by pernielsentikaer** [comment]
Hi @micnncim 👋 

Did you change anything in your end regarding API calls that might cause a “422 Unprocessable Entity” instead of “401 Unauthorized”

Error 422 should be: The contents of the request contains errors, such as required fields that are missing or validation errors 

Could you try this yourself with a wrong token to check this, does it also happen for you now?

**Comment by fe9lix** [comment]
Closing this PR due to inactivity – feel free to reopen later.

**Comment by micnncim** [comment]
Couldn't work on this as I was busy. I'll fix issues and reopen this PR in a few days.

**Pull request closed without merging**